### ISSUE TITLE
isNaN --> Number.isNan

### ISSUE THREAD

**Issue by stevenvachon** [eslint/eslint#10313]
**Please describe what the rule should do:**
Should push us to use `Number.isNaN` instead of `isNaN`.

**What category of rule is this? (place an "X" next to just one item)**

[ ] Enforces code style
[ ] Warns about a potential error
[x] Suggests an alternate way of doing something
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
isNaN(1);
isNaN(NaN);
```

**Why should this rule be included in ESLint (instead of a plugin)?**
It's native ES2015

See #375

**Comment by platinumazure** [comment]
Hi @stevenvachon, thanks for the issue.

This can already be done using [`no-restricted-globals`](https://eslint.org/docs/rules/no-restricted-globals). You can even include a message asking the user to use `Number.isNaN`.

I don't really want to add a rule in ESLint core just for this case-- we already have too many. But you are more than welcome to use or create a plugin with this rule.

**Comment by stevenvachon** [comment]
Ok, thank you.
